src/System.IO.FileSystem/src/System/IO/DirectoryInfo.cs

```diff
@@ -430,20 +427,23 @@ public void Delete(bool recursive)
             FileSystem.Current.RemoveDirectory(FullPath, recursive);
         }
 
-        // Returns the fully qualified path
+        /// <summary>
+        /// Returns the original path. Use FullPath or Name properties for the path / directory name.
+        /// </summary>
         public override String ToString()
         {
             return DisplayPath;
         }
 
-        private static String GetDisplayName(String originalPath, String fullPath)
+        private static String GetDisplayName(String originalPath)
         {
             Debug.Assert(originalPath != null);
-            Debug.Assert(fullPath != null);
 
+            // Desktop documents that the path returned by ToString() should be the original path.
+            // For SL/Phone we only gave the directory name regardless of what was passed in.
```

Comment:
Figured it would be good to document where it differed historically. Might clarify why it was different here to start with.